src/server/pkg/storage/fileset/index/writer.go

```diff
@@ -151,12 +151,20 @@ func (w *Writer) Close() error {
 	if err != nil {
 		return err
 	}
-	if _, err := pbutil.NewWriter(objW).Write(w.root); err != nil {
+	defer func() {
+		if retErr == nil {
```

Comment:
This seems like a bug in that if we encounter another error we won't call `objW.Close()` and will keep an object write open.

Reply:
That is what we want. We want the object storage provider to timeout the write and delete what was written. If we close it, the data written will be kept. There might be a better pattern for this though that is not as weird (and potentially problematic for certain object storage providers).

Reply:
I see, this makes sense, I think there is a better pattern here which is to create a `ctx` for the `Writer` and then explicitly call `cancel` if there's an error, that'll free up the resources immediately, and make it a little more clear that not calling `Close` is intentional in that case.

Reply:
After thinking about this more, and looking at the object storage clients, I decided to go ahead and always close. The object storage clients handle this case correctly. Also, when we change the upload interface to operate on a passed in reader, this potential problem will go away.